src/Form/LegacyConsumer/Commands/SetupFieldPersistance.php

```diff
@@ -21,7 +21,15 @@ function( $donationID, $donationData ) use ( $hook ) {
 					function( $field ) use ( $donationID ) {
 						if ( isset( $_POST[ $field->getName() ] ) ) {
 							$value = wp_strip_all_tags( $_POST[ $field->getName() ], true );
-							give_update_payment_meta( $donationID, $field->getName(), $value );
+
+							if ( $field->shouldStoreAsDonationMeta() ) {
+								give_update_payment_meta( $donationID, $field->getName(), $value );
+							}
+
+							if ( $field->shouldStoreAsDonorMeta() ) {
+								$donorID = give_get_payment_meta( $donationID, '_give_payment_donor_id' );
+								Give()->donor_meta->update_meta( $donorID, $field->getName(), $value );
+							}
```

Comment:
I would like to make this a bit more generic to make it easy for folks to provide their own storage methods:
```suggestion
							switch($field->storageType) {
								case Field::DONATION_STORAGE_TYPE:
									give_update_payment_meta( $donationID, $field->getName(), $value );
									break;
									
								case Field::DONOR_STORAGE_TYPE:
									$donorID = give_get_payment_meta( $donationID, '_give_payment_donor_id' );
									Give()->donor_meta->update_meta( $donorID, $field->getName(), $value );
									break;
								
								default:
									do_action('give_store_form_field', $field->storageType, $field, $value);
									do_action("give_store_form_field_{$field->storageType}", $field, $value);
							}
```
Something like that, anyway. What do you think?

Reply:
I don't think that generalizing it is necessary, as this is already a convenience added on top of the existing `give_insert_payment` action. The current approach provides a fluent, declarative path for storing as the most common donor or donation meta. If they wanted to handle it themselves then the hooks already exist for that.

Reply:
Also, I have separated the donor and donation meta flags as separate properties so that both can be used. I think it would make sense to still want the data collected by the donation to exist on the donation for record keeping while also storing the meta on the donor.

Reply:
Resolved in a call, see https://github.com/impress-org/givewp/pull/5652#issuecomment-788260103.